src/touchable.tsx

```diff
@@ -52,56 +44,80 @@ interface CommonProps {
  * this way we can know the type of the union by checking that property
  * See https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions
  */
-export interface PropsHref extends CommonProps {
+
+interface PropsHref extends CommonProps {
+    maybe?: undefined;
     href: string;
     newTab?: boolean;
     loadOnTop?: boolean;
     to?: undefined;
     onPress?: undefined;
+    /** with "href", onNavigate will be executed before the navigation */
+    onNavigate?: () => void | Promise<void>;
 }
-export interface PropsOnPress extends CommonProps {
-    onPress: PressHandler;
-    href?: undefined;
+
+interface PropsMaybeHref extends CommonProps {
+    maybe: true;
+    href?: string;
+    newTab?: boolean;
+    loadOnTop?: boolean;
     to?: undefined;
-    formId?: string;
+    onPress?: undefined;
+    /** with "href", onNavigate will be executed before the navigation */
+    onNavigate?: () => void | Promise<void>;
 }
 
-export interface PropsTo extends CommonProps {
+interface PropsTo extends CommonProps {
+    maybe?: undefined;
     to: string | Location;
     fullPageOnWebView?: boolean;
     replace?: boolean;
     href?: undefined;
     onPress?: undefined;
+    /** with "to", onNavigate will be executed in parallel to the navigation */
+    onNavigate?: () => void | Promise<void>;
 }
-export interface PropsMaybeHref extends CommonProps {
-    maybe: true;
-    href?: string;
-    newTab?: boolean;
-    loadOnTop?: boolean;
-    to?: undefined;
-    onPress?: undefined;
-}
-export interface PropsMaybeTo extends CommonProps {
+
+interface PropsMaybeTo extends CommonProps {
     maybe: true;
     to?: string | Location;
     fullPageOnWebView?: boolean;
     replace?: boolean;
     href?: undefined;
     onPress?: undefined;
+    /** with "to", onNavigate will be executed in parallel to the navigation */
```

Comment:
As far as I can see, it's not in parallel, right? You wait for the promise to be resolved, and after that you navigate

Reply:
what you comment is true in the "href", not for the "to".

The "to" prop will use a `<Link>` element which produces a client side navigation, so it is ok to perform the onNavigate actions in parallel (this was already happening in current implementation)